src/protocols/ethereum/libs/EthereumAdapter.ts

```diff
@@ -112,24 +135,51 @@ export class EthereumAdapter extends BaseProtocolAdapter {
     return ETH_NETWORK_DEFAULT_SETTINGS[networkType];
   }
 
-  override async fetchBalance() {
-    return ''; // TODO
+  override async fetchBalance(address: string): Promise<string> {
+    const { ethActiveNetworkSettings } = useEthNetworkSettings();
+    const { nodeUrl } = ethActiveNetworkSettings.value;
+    const web3Eth = new Web3Eth(nodeUrl);
+    const balanceInWei = await getBalance(web3Eth, address, 'latest', NUMBER_DATA_FORMAT);
+    return fromWei(balanceInWei, 'ether').toString();
   }
 
   override isAccountAddressValid(address: string) {
     return isAddress(address);
   }
 
-  override async isAccountUsed() {
-    return false; // TODO
+  override async isAccountUsed(address: string): Promise<boolean> {
+    const balance = parseFloat(await this.fetchBalance(address));
+    if (balance > 0) {
+      return true;
+    }
+    // in case of 0 balance, check if account has transaction history
+    const txCount = await this.getTransactionCount(address);
+    return txCount > 0;
```

Comment:
We can use one request with all the needed info (balance and tx count in one place).
https://www.quicknode.com/docs/ethereum/eth_getAccount

Reply:
eth_getAccount is not part of the standard interface for Ethereum and many nodes don't have this function
https://ethereum.github.io/execution-apis/api-documentation/

But we can make both requests at the same time to save time

Reply:
I also changed the node which we are using for the mainnet, and recovery now only takes ~6 seconds (for all protocols)